### PR TITLE
chore: [IOPID-359] First onboarding refactor

### PR DESCRIPTION
**Pull request by SabPicariello**
## Short description
Include a summary of the changes.

## List of changes proposed in this pull request
- ts/sagas/profile.ts: last_app_version and preferred_languages are always set if not found at every profile upsert. Upsert and checks on the profile are prevented until acceptance of the tos which becomes the first upsert of the app (during this first upsert it also sets the app version and language). This way there are also no more upsert and duplicate checks when creating a new profile.
- ts/store/reducers/profile.ts: Having profile version 0 is not enough to tell if you are an onboarding. The check is now done on service_preferences_settings which is set to legacy until the last onboarding screen.
- *.test.ts: updates

## How to test
Run dev server with first onboarding set to true and make sure that all the described behaviors are respected.

### CODE REVIEW

**Comment by thisisjp** [comment]
@SabPicariello 
>since the English language has all the keys of the Italian language

this is not true: content designers always provide the Italian localization. We're planning to better structure the localization workflow, but in any case the translated strings will arrive t+1.

cc @AliceAzzolini

**Comment by SabPicariello** [comment]
> 



> @SabPicariello
> 
> > since the English language has all the keys of the Italian language
> 
> this is not true: content designers always provide the Italian localization. We're planning to better structure the localization workflow, but in any case the translated strings will arrive t+1.
> 
> cc @AliceAzzolini

I am aware of it. I meant that any key present in the Italian file, is also present in the English file (although in this one we sometimes find values associated with keys, in Italian language). This allows us to move the fallback language (in case the user has, for example, the device in French) from Italian to English, without creating errors. 
The fact that the user will see some sentences in Italian is certain, but a few sentences are always better than the whole application. Do you agree?

File: ts/sagas/profile.ts [hunk 1]
```diff
@@ -121,7 +118,18 @@ function* createOrUpdateProfileSaga(
...
   const rawAppVersion = yield* call(getAppVersion);
-  const maybeAppVersion = O.fromEither(AppVersion.decode(rawAppVersion));
+
+  const appVersion = pipe(
+    O.fromEither(AppVersion.decode(rawAppVersion)),
+    O.fold(
+      () => undefined,
+      version => version
+    )
+  );
```
Comment (shadowsheep1) [inline ts/sagas/profile.ts:128]:
```suggestion
const appVersion = pipe(
    rawAppVersion,
    AppVersion.decode,
    E.fold(
      () => undefined,
      version => version
    )
  );
```
What do you think of this version?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
Just a minor change.

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**